ext/nnstreamer/tensor_source/tensor_src_tizensensor.c

```diff
@@ -349,7 +384,122 @@ static void
+  int ret = 0;
+
+  switch (prop_id) {
+  case PROP_SILENT:
+    self->silent = g_value_get_boolean (value);
+    silent_debug ("Set silent = %d", self->silent);
+    break;
+  case PROP_TYPE:
+    {
+      sensor_type_e new_type = g_value_get_enum (value);
+
+      _LOCK (self);
+
+      if (new_type != self->type) {
+        /* Different sensor is being used. Clean it up! */
+        ret = _ts_clean_up_handle (self);
+
+        if (ret) {
+          GST_ELEMENT_ERROR (self, TIZEN_SENSOR, FAILED,
+              ("Calling _ts_clean_up_handle at set PROP_TYPE has failed."),
+              ("_ts_clean_up_handle() returns %d", ret));
+        }
+
+        silent_debug ("Set type from %d --> %d.", self->type, new_type);
+        self->type = new_type;
+      } else {
+        silent_debug ("Set type ignored (%d --> %d).", self->type, new_type);
+      }
+
+      _UNLOCK (self);
+    }
+    break;
+  case PROP_SEQUENCE:
+    {
+      guint new_sequence = g_value_get_uint (value);
+
+      _LOCK (self);
+
+      if (self->sequence != new_sequence) {
+        /* Different sensor is being used. Clean it up! */
+        ret = _ts_clean_up_handle (self);
+
+        if (ret) {
+          GST_ELEMENT_ERROR (self, TIZEN_SENSOR, FAILED,
+              ("Calling _ts_clean_up_handle at set PROP_SEQUENCE has failed."),
+              ("_ts_clean_up_handle() returns %d", ret));
+        }
+
+        silent_debug ("Set sequence from %u --> %u.", self->sequence,
+            new_sequence);
+        self->sequence = new_sequence;
+      } else {
+        silent_debug ("Set sequence ignored (%u --> %u).", self->sequence,
+            new_sequence);
+      }
+
+      _UNLOCK (self);
+    }
+    break;
+  case PROP_MODE:
+    {
+      sensor_op_modes new_mode = g_value_get_enum (value);
+
+      _LOCK (self);
+
+      if (new_mode != self->mode) {
+        silent_debug ("Set mode from %d --> %d.", self->mode, new_mode);
+        self->mode = new_mode;
+
+        /* Same sensor is kept. Only mode is changed */
+        ret = _ts_reconfigure (self);
+
+        if (ret) {
+          GST_ELEMENT_ERROR (self, TIZEN_SENSOR, FAILED,
+              ("Calling _ts_reconfigure at set PROP_MODE has failed."),
+              ("_ts_reconfigure () returns %d", ret));
+        }
+
+      } else {
+        silent_debug ("Set mode ignored (%d --> %d).", self->mode, new_mode);
+      }
+
+      _UNLOCK (self);
+    }
+    break;
+  case PROP_FREQ:
+    {
+      gint n = self->freq_n;
+      gint d = self->freq_d;
+
+      _LOCK (self);
+
+      self->freq_n = gst_value_get_fraction_numerator (value);
+      self->freq_d = gst_value_get_fraction_denominator (value);
+
+      silent_debug ("Set operating frequency %d/%d --> %d/%d",
+          n, d, self->freq_n, self->freq_d);
+
+      if (n != self->freq_n || d != self->freq_d) {
+        /* Same sensor is kept. Only mode is changed */
+        ret = _ts_reconfigure (self);
```

Comment:
same as above, reset freq values if reconfigure fails (This will keep the `get_property` consistent).